toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -1476,7 +1497,7 @@ func createUserWithPassword(installChroot *safechroot.Chroot, user configuration
 		}
 		isRoot = true
 	} else {
-		err = userutils.AddUser(user.Name, hashedPassword, user.UID, installChroot)
+		err = userutils.AddUser(user.Name, user.HomeDirectory, user.PrimaryGroup, hashedPassword, user.UID, installChroot)
```

Comment:
If you are going to set the user's primary group here, then you should skip the `ConfigureUserGroupMembership` call in `addUsers` when the user is added (as opposed to the user being updated).

Reply:
Removed and renamed function to `ConfigureUserSecondaryGroupMembership()`.

Reply:
Actually... that might break MIC...

Reply:
Split into `ConfigureUserPrimaryGroupMembership()` and `ConfigureUserSecondaryGroupMembership()` so we can target what we need.